### DISCUSSION TITLE
IOS records not being able to be read past a few seconds

### DISCUSSION THREAD

**Discussion by CopBox** [openai/whisper#1678] [Q&A]
Sorry for the bother but I was wondering if anyone here would be able to help, I have tried using polyfil in the frontend to help with the audio, I checked audio inputs and they are coming in crisp and full, but when they are sent over to the API it only reads the first few seconds,  can someone tell me where I am going wrong? Been at it for a few days and narrowed it down to the API understanding the audio, it works with desktop records and even youtube videos but this doesn't seem to work with any iphone recordings I have done:

app.post('/audio', upload.single('audio'), async (req, res, next) => {
  try {
      const audioFile = req.file;
      console.log(audioFile);
      
      if (!audioFile) {
          return res.status(400).json({ error: 'No audio file uploaded' });
      }

      // Validate MIME type of audio file
      const validMimeTypes = ['audio/wav', 'audio/mp3', 'audio/webm', 'audio/mp4'];
      if (!validMimeTypes.includes(audioFile.mimetype)) {
          return res.status(400).json({ error: 'Unsupported audio format' });
      }

      // Adjust filename based on mimetype
      let fileName;
      switch(audioFile.mimetype) {
          case 'audio/wav':
              fileName = 'audio.wav';
              break;
          case 'audio/mp3':
              fileName = 'audio.mp3';
              break;
              case 'audio/mp4':
                fileName = 'audio.mp4';
                break;
          case 'audio/webm':
              fileName = 'audio.webm';
              break;
          default:
              throw new Error('Unsupported audio format');
      }

      const formData = new FormData();
      formData.append('file', audioFile.buffer, { filename: fileName, contentType: audioFile.mimetype });
      formData.append('model', 'whisper-1');
      formData.append('response_format', 'json');

      const config = {
          headers: {
              ...formData.getHeaders(),
              'Authorization': `Bearer KEY_OBV_NOT_POSTING_HERE`
          },
      };

      // Send the audio data to Whisper ASR system
      const response = await axios.post('https://api.openai.com/v1/audio/transcriptions', formData, config);
      
      // Check if the response contains the expected data
      if (!response.data || !response.data.text) {
          throw new Error("Unexpected response format from Whisper API");
      }

      const transcribedText = response.data.text;
      // Send the transcribed text back to the client
      res.json({ transcription: transcribedText });

  } catch (error) {
      // Log detailed error information
      console.error("Error sending data to Whisper API:", error.message, error.response ? error.response.data : "");
      res.status(500).json({ error: 'Error transcribing audio' });
  }
});

**Comment by phineas-pta** [comment]
your question should be asked at: https://community.openai.com/tag/whisper
